Code/Framework/AzCore/AzCore/Script/ScriptAdapterComponent.cpp

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright (c) Huawei Technologies Co., Ltd. 2021-2021. All rights reserved.
+ * Description: ScriptAdapterComponent.
+ * Create: 2021-06-28
+ */
+#include <AzCore/Script/ScriptAdapterComponent.h>
+#include <AzCore/Script/ScriptSystemBus.h>
+#include <AzCore/Script/lua/lua.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/RTTI/RTTI.h>
+#include <AzCore/RTTI/BehaviorContextUtilities.h>
+#include <AzCore/Component/ComponentApplication.h>
+#include <AzCore/StringFunc/StringFunc.h>
+#include <AzCore/Utils/Utils.h>
+#include <AzCore/IO/SystemFile.h>
+#if defined(WIN32) || defined(WIN64)
+#include <atlsecurity.h>
+#endif
+
+namespace AZ
+{
+    const char* ScriptAdapterComponent::RELATIVE_LUA_PATH = "gen\\LuaAPI\\";
+    const char* ScriptAdapterComponent::GEN_DIR_NAME = "gen";
+    const char* ScriptAdapterComponent::GLOBAL_LUA_FILE_NAME = "GlobalsStaticLuaData.lua";
+    const char* ScriptAdapterComponent::CLASS_LUA_FILE_NAME = "ClassesStaticLuaData.lua";
+    const char* ScriptAdapterComponent::EBUS_LUA_FILE_NAME = "EbusesStaticLuaData.lua";
+    const int ScriptAdapterComponent::NAME_BUFFER_SIZE = 64;
```

Comment:
This implementation is Windows specific and would need to be Platform abstracted to allow for implementations on for the other supported platforms.

Pathwise using backslashes should be avoided and a dedicated path class like [AZ:IO::Path/AZ::IO::FixedMaxPath/AZ::IO::PathView](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h) should be used.

Also constant variables names would need to be updated to fit within the coding standard of [UpperCamelCase](https://github.com/o3de/sig-core/blob/main/governance/Coding-Standards-and-Style-Guide.md#constant-names)